synapse/storage/databases/main/schema/delta/58/24drop_event_json_index.sql

```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- this index is essentially redundant. The only time it was ever used was when purging
+-- rooms - and Synapse 1.24 will change that.
```

Comment:
Is there any reason to leave this and drop it in a future version? (E.g. in case someone upgrades to 1.24, then rolls back to 1.23?) This only being used for purging events means it probably doesn't really matter though. 🤷

Reply:
indeed. The fact that matrix.org is fine without it makes me think that we don't need to worry too much.